library/src/scripts/components/forms/select/Tokens.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import Select from "react-select";
+import { getRequiredID, IOptionalComponentID } from "@library/componentIDs";
+import classNames from "classnames";
+import { t } from "@library/application";
+import menuList from "@library/components/forms/select/overwrites/menuList";
+import menu from "@library/components/forms/select/overwrites/menu";
+import selectContainer from "@library/components/forms/select/overwrites/selectContainer";
+import doNotRender from "@library/components/forms/select/overwrites/doNotRender";
+import Paragraph from "@library/components/Paragraph";
+import { IComboBoxOption } from "./BigSearch";
+import selectOption from "@library/components/forms/select/overwrites/selectOption";
+import valueContainerTokens from "@library/components/forms/select/overwrites/valueContainerTokens";
+import multiValueContainer from "./overwrites/multiValueContainer";
+import multiValueLabel from "./overwrites/multiValueLabel";
+import multiValueRemove from "./overwrites/multiValueRemove";
+
+interface IProps extends IOptionalComponentID {
+    label: string;
+    labelNote?: string;
+    disabled?: boolean;
+    className?: string;
+    placeholder?: string;
+    options: IComboBoxOption[];
+    setAuthor: (authors: IComboBoxOption[]) => void;
+}
+
+/**
+ * Implements the search bar component
+ */
+export default class Tokens extends React.Component<IProps> {
+    private id: string;
+    private prefix = "tokens";
+    private inputID: string;
+
+    constructor(props: IProps) {
+        super(props);
+        this.id = getRequiredID(props, this.prefix);
+        // this.searchButtonID = this.id + "-searchButton";
+        this.inputID = this.id + "-tokenInput";
+    }
+
+    private handleOnChange = (newValue: any, actionMeta: any) => {
+        this.props.setAuthor(newValue);
+    };
+
+    public render() {
+        const { className, disabled, options } = this.props;
+
+        /** The children to be rendered inside the indicator. */
+        const componentOverwrites = {
```

Comment:
I'd like to see the same thing as mentioned in the `SelectOne` feedback. Let's pull out everything we can that is not JSX here into class getters, properties, or methods.